tests/src/com.github.ben-manes.caffeine/caffeine/2.9.3/src/test/java/com_github_ben_manes_caffeine/caffeine/PopulationTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_github_ben_manes_caffeine.caffeine;
+
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.github.benmanes.caffeine.cache.AsyncLoadingCache;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PopulationTest {
+    @Test
+    void testManual() {
+        Cache<String, String> cache = Caffeine.newBuilder().expireAfterWrite(10, TimeUnit.MINUTES).maximumSize(10_000).build();
+        assertThat(cache.getIfPresent("Hello")).isNull();
+        assertThat(cache.get("Hello", k -> "World")).isEqualTo("World");
+        cache.put("Hello", "World");
+        assertThat(cache.getIfPresent("Hello")).isEqualTo("World");
+        cache.invalidate("Hello");
+        assertThat(cache.getIfPresent("Hello")).isNull();
+    }
+
+    @Test
+    void testLoading() {
+        LoadingCache<String, String> cache = Caffeine.newBuilder().maximumSize(10_000).expireAfterWrite(10, TimeUnit.MINUTES)
```

Comment:
The only difference (that I can spot) between this test and [testRefresh](https://github.com/oracle/graalvm-reachability-metadata/pull/388/files#diff-220cd86ae77195bc08dc7249346abd6ccbbbd4ac5f7b828f0bd266b128cc25baR26) test is usage of `expireAfterWrite` instead of `refreshAfterWrite` function. Besides that, `expireAfterWrite` functionality was exercised in [this test](https://github.com/oracle/graalvm-reachability-metadata/pull/388/files#diff-ce2f726eecc1294577266412d9aa4abe59eea73288888fa23df5e9daf6d2ed5cR21), so I believe this test doesn't test anything new and therefore can be removed, right?

Reply:
- `testLoading()` is now renamed to `testExpireAfterWrite()` and uses logical assertions.